### ISSUE TITLE
Unused patches cause build to fail

### ISSUE THREAD

**Issue by barsoosayque** [ipetkov/crane#576]
For our project we use a git dependency of [bevy](https://github.com/bevyengine/bevy) (to be precise, it's our own fork w/ a specific branch), and bevy repository is basically a monorepo with a bunch of smaller crates. Some of our project's other dependencies can themselves depend on either bevy (the big crate), or depend or smal individual bevy_* crates. 

Because of that, I had to make patches for both `bevy` and each and every of its smaller crates. Some of them are left unused, which is highlighted in cargo (when vendoring, for example):
```
warning: Patch `bevy_audio v0.12.1 (https://github.com/barsoosayque/bevy?branch=gltf-skin-0.12#dc13ac5b)` was not used in the crate graph.
Patch `bevy_dylib v0.12.1 (https://github.com/barsoosayque/bevy?branch=gltf-skin-0.12#dc13ac5b)` was not used in the crate graph.
<...>
```
And naturally it's also reflected in `Cargo.lock`:
```toml
[[patch.unused]]
name = "bevy_audio"
version = "0.12.1"
source = "git+https://github.com/barsoosayque/bevy?branch=gltf-skin-0.12#dc13ac5bd81908a9199fdf3549061f4320b0b478"

[[patch.unused]]
name = "bevy_dylib"
version = "0.12.1"
source = "git+https://github.com/barsoosayque/bevy?branch=gltf-skin-0.12#dc13ac5bd81908a9199fdf3549061f4320b0b478"

# <...>
```

Building it like that results in an error:
```
++ command cargo check --release --target x86_64-pc-windows-gnu
error: failed to resolve patches for `https://github.com/rust-lang/crates.io-index`

Caused by:
   patch for `bevy_audio` in `https://github.com/rust-lang/crates.io-index` failed to resolve

Caused by:
   The patch location `https://github.com/barsoosayque/bevy?branch=gltf-skin-0.12` does not appear to contain any packages matching the name `bevy_audio`.
```

The easy fix is to comment out unused patches, which would then build just fine. I wasn't sure if it's my own silly mistakes, an internal crane error, or some sort of lock file/vendoring issue.

**Comment by ipetkov** [comment]
Interesting, I wonder if somehow the unused patch dependency is not being vendored... Do you have a(n ideally minimal) reproduction I can check out and investigate further?

**Comment by barsoosayque** [comment]
I'll try to make a repository with minimal repro this weekend, shouldn't be too hard

**Comment by barsoosayque** [comment]
I have created a small repository with a repro: [barsoosayque/crane-unused-repro](https://github.com/barsoosayque/crane-unused-repro). But to be fair, in my minimal reproduction I have a different error:

```
error: builder for '/nix/store/a44dk4kzy1apyr4z8zdmdrnxr8xqgq0w-crane-unused-repro-deps-master.drv' failed with exit code 101;
       last 10 log lines:
       > Caused by:
       >   failed to clone into: /build/source/.cargo-home/git/db/image-c1c0bf49fde10069
       >
       > Caused by:
       >   network failure seems to have happened
       >   if a proxy or similar is necessary `net.git-fetch-with-cli` may help here
       >   https://doc.rust-lang.org/cargo/reference/config.html#netgit-fetch-with-cli
       >
       > Caused by:
       >   failed to resolve address for github.com: Temporary failure in name resolution; class=Net (12)
       For full logs, run 'nix log /nix/store/a44dk4kzy1apyr4z8zdmdrnxr8xqgq0w-crane-unused-repro-deps-master.drv'.
error: 1 dependencies of derivation '/nix/store/129mxpckqizms2knwirw4h46jaj6n5mj-crane-unused-repro-master.drv' failed to build
``` 

Which only happens if there is an unused override for a dependency (in this case, `image` lib). If I remove the one unused patch, it builds fine. Also, manual `cargo build` has no issues with building. And even more than that, there are two patches with a github link, but only the unused one creates an error.

**Comment by ipetkov** [comment]
Thanks for the reproduction! I found the issue (which I also overlooked in the initial description, so apologize for that): cargo will put unused patch deps under `[[patch.unused]]` instead of `[[package]]` in `Cargo.lock`. We just need to take those entries into account when vendoring crate sources!

Should be fixed in #581
